marine/marine.py

```diff
@@ -164,24 +172,40 @@ def filter_and_parse(
         self._marine.marine_free(marine_result)
         return success, result
 
+    def _resolve_err_msg(self, err_msg: POINTER(POINTER(c_char))) -> Optional[str]:
+        if not err_msg.contents:
+            return None
+        error = string_at(err_msg.contents).decode("utf-8")
```

Comment:
Can `string_at` or `decode` fail for some reason? If so, should we put it in `try finally` to make sure the contents are always freed?

Reply:
`string_at` should not fail, since is marine core is guaranteed to provide a valid string there.
I guess it would be good to move the `decode` after the free though.